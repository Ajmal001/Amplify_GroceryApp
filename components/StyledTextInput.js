import React from "react";
import { TextInput, StyleSheet } from "react-native";
import { grey } from "../utils/colors";
import PropTypes from "prop-types";

export default function StyledTextInput({ value, placeholder, onChangeText }) {
  return (
    <TextInput
      style={styles.textInput}
      onChangeText={onChangeText}
      value={value}
      placeholder={placeholder}
    />
  );
}

const styles = StyleSheet.create({
  textInput: {
    height: 40,
    borderColor: grey,
    borderWidth: 2,
    borderRadius: 10,
    paddingLeft: 15,
    marginTop: 10,
    marginLeft: 30,
    marginRight: 30,
  },
});

StyledTextInput.propTypes = {
  onChangeText: PropTypes.func.isRequired,
  value: PropTypes.string.isRequired,
  placeholder: PropTypes.string,
};
