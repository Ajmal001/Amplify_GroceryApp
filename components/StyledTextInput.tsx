import React from "react";
import { TextInput } from 'react-native-paper';
import { secondaryColor } from "../utils/helpers";


interface Props {
  onChangeText: () => void;
  value: string;
  placeholder?: string;
  multiline?: boolean;
  keyboardType?: any;
  style?: object;
  label: string;
}

// const StyledTextInput: React.FC<Props>
const StyledTextInput: React.FC<Props> = ({
  value, placeholder, onChangeText, label, multiline = false, keyboardType = 'default', style = {}
}) => {

  return (
    <TextInput
      label={label}
      mode="outlined"
      onChangeText={onChangeText}
      value={value}
      placeholder={placeholder}
      testID="TextInput"
      multiline={multiline}
      keyboardType={keyboardType}
      theme={{
        colors: {
          primary:secondaryColor
        },
      }}
      style={style}
    />
  );
}

export default StyledTextInput