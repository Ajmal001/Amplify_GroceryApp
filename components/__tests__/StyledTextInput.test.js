import React from 'react';
import renderer from 'react-test-renderer';
import StyledTextInput from '../StyledTextInput';


test('renders correctly', () => {
  const tree = renderer.create(<StyledTextInput
    value={''} 
    placeholder={''}
    onChangeText={x => x} />).toJSON();
  expect(tree).toMatchSnapshot();
});