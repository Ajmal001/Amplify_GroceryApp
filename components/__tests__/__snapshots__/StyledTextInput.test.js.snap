// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<TextInput
  allowFontScaling={true}
  onChangeText={[Function]}
  placeholder=""
  rejectResponderTermination={true}
  style={
    Object {
      "borderColor": "#5c626b",
      "borderRadius": 10,
      "borderWidth": 2,
      "height": 40,
      "marginLeft": 30,
      "marginRight": 30,
      "marginTop": 10,
      "paddingLeft": 15,
    }
  }
  underlineColorAndroid="transparent"
  value=""
/>
`;
